### DISCUSSION TITLE
Raw HTML export

### DISCUSSION THREAD

**Discussion by rikhuijzer** [fonsp/Pluto.jl#1607] [Pluto development] [answered]
Now that Pluto has built in package management, it is almost too good. So, my compliments on that :+1: What is still a bit of a problem for me is that I couldn't find a way yet to cleanly get the Pluto code and output into my blog. So, I've created a package: [PlutoHTML.jl](https://github.com/rikhuijzer/PlutoHTML.jl).

This package provides one method, namely "notebook2html". This method takes a notebook path or a `Pluto.Notebook` object and 

- evaluates the notebook via Pluto, 
- reads the outputs from Pluto, and
- converts the outputs to HTML.

@fonsp and other developers, is this functionality something that you would like to move into the Pluto.jl codebase, or should it stay a stand-alone package? It's not too much code.

EDIT: Example post created via PlutoHTML here: https://huijzer.xyz/posts/frequentist-bayesian-coin-flipping/. A link to the notebook is at the bottom of the page.

**Comment by fonsp** [comment, marked as answer]
Hoi Rik!

That looks really nice! Happy to see that you played around with Pluto's internals.

Right now, I am in a phase where I am super excited about everything that we can do with the existing built-in HTML export, which includes the full Pluto frontend. This lets us open a binder in the background, run a plutosliderserver and [more](https://github.com/fonsp/Pluto.jl/pull/1623), and it means that the export is truly WYSIWYG.

On the other hand, there are some disadvantages to this technique:
- Only the most modern browsers are supported by Pluto, you see a blank screen otherwise
- There is some performance impact, because content is loaded dynamically
- The result is more difficult to style than a traditional HTML snippet. You currently need to inset the notebook using an iframe ([like here](https://computationalthinking.mit.edu/)) and insert styles within the notebook itself...

Your package solves these three issues! Eventually, I think we can solve the first two without changing the format, e.g. using [SSR on our preact app](https://preactjs.com/guide/v10/server-side-rendering/), but this package is a good solution for all three right now.

About the name: PlutoHTML could lead to confusion, since Pluto already has an HTML export system built-in (`PlutoSliderServer.export_notebook` makes it easier to use from julia). Maybe the name could reflect the different technique, like PlutoStaticHTML, PlutoSimpleHTML, PlutoOfflineHTML or PlutoSSR?

The blog article looks really nice! I will try it out soon!

**Comment by rikhuijzer** [comment]
> Right now, I am in a phase where I am super excited about everything that we can do with the existing built-in HTML export, which includes the full Pluto frontend. This lets us open a binder in the background, run a plutosliderserver and more, and it means that the export is truly WYSIWYG.

Agreed. The HTML export definitely has it's problems too. Thank you for also seeing the benefits.

> Maybe the name could reflect the different technique, like PlutoStaticHTML, PlutoSimpleHTML, PlutoOfflineHTML or PlutoSSR?

Great suggestion. Done :heavy_check_mark:

Thank you for your feedback and being open about this package. Feel free to let me know if you have comments or suggestions at a later point. For now, I'll keep working on this package and maybe even register it at some point. :+1:

**Comment by AbhimanyuAryan** [comment]
@rikhuijzer looks like it doesn't hide the hidden Pluto notebook cells. Is it possible to hide cells that are hidden in Pluto.jl?

**Reply by rikhuijzer** [reply]
> @rikhuijzer looks like it doesn't hide the hidden Pluto notebook cells. Is it possible to hide cells that are hidden in Pluto.jl?

Probably yes. Could you open an issue at the repo? I’m on my phone and in a hurry now

**Reply by AbhimanyuAryan** [reply]
sure
